designs/function-node-lifecycle/README.md

```diff
@@ -41,17 +41,100 @@ Following figure shows new lifecycle model of function node.  While creating a f
 
 ### Use of External JavaScript Libraries
 
-New config tab includes declaration of list of JavaScript variable and required module.
-
-![config](./config.png)
-
 If required modules are not installed, we need a means to install NPM modules from inside Node-RED editor.  This requires considerations on security, restriction of environments, user interaction, and other topics.  In the above lifecycle model, we propose to make automatic (e.g. for headless Node-RED) or interactive module installation selectable by changing a value in `settings.js`.  
 
-Function node declares its required NPM module list to runtime by using new API.   For interactive mode, we add NPM management UI in editor settings panel.
+Function node declares its required NPM module list in its settings panel.   For interactive mode, users can install, uninstall, or update NPM module from this settings panel.
```

Comment:
Can we have some discussion over what runtime APIs must now be exposed to support these functions? Are they functions specific to the Function node (so provided by the Function node itself), or a new `modules` api of the runtime itself.

Reply:
Since the Node-RED runtime may perform the installation in current design, it is natural for the runtime to provide the functionality.  So, added a proposal to extend existing `RED.require` function.

Reply:
Sorry for the late reply. I have fixed the API as indicated.
I may not have understood the pointing correctly, but the Node API only exposes the `require` function to nodes, so the original proposal was designed to extend the `require` function.  
Fixed to expose `addModule`, `remodeModule`, and `getModuleInfo` as node API.